eks/terraform/modules/cluster/main.tf

```diff
@@ -335,7 +260,7 @@ resource "aws_eks_addon" "csi-driver" {
 resource "aws_eks_addon" "vpc-cni" {
   cluster_name             = aws_eks_cluster.cluster.name
   addon_name               = "vpc-cni"
-  service_account_role_arn = module.vpc_cni_irsa_role.iam_role_arn
+  service_account_role_arn = var.use_irsa_v1 ? module.vpc_cni_irsa_role.iam_role_arn : null
```

Comment:
@Luay-Sol I recommend to also add `addon_version` argument here, otherwise it will use default version for the EKS cluster version.

The pod identity does not work if the `vpc-cni` version is earlier than `1.15.5-eksbuild.1`. If we don't set the `addon_version` explicitly here, the default version for `1.27` for example, is `v1.15.1-eksbuild.1` which is earlier than the required version.

Reply:
@nagsubhrajitt do you think it is required here. Considering 1.27 it is going to be out of regular support in couple day from now (July 24).

Another consideration is this project is supposed to be used for cluster creation of latest version of k8s/eks. So I don't see a situation or a reason for supporting older versions. 

Reason I am on the fence about add-ons version hard coding: it is going to be challenging for us (as maintainers) for this project as well as for the users.